packages/fuel-indexer-macros/src/helpers.rs

```diff
@@ -670,59 +575,126 @@ pub fn unwrap_or_default_tokens(field_typ_name: &str, nullabel: bool) -> TokenSt
     }
 }
 
-/// Get tokens for a field's `.to_bytes()`.
-pub fn to_bytes_tokens(field_typ_name: &str) -> TokenStream {
-    if EXTERNAL_FIELD_TYPES.contains(field_typ_name) {
-        to_bytes_method_for_external_type(field_typ_name)
-    } else if !ASREF_BYTE_TYPES.contains(field_typ_name) {
-        quote! { .to_le_bytes() }
-    } else {
-        quote! {}
+/// Get tokens for a given field type's `.to_bytes()`.
+pub fn to_bytes_tokens(
+    field_typ_name: &str,
+    processed_type_result: &ProcessedFieldType,
+) -> TokenStream {
+    let ProcessedFieldType { base_type, .. } = &processed_type_result;
+    match base_type {
+        FieldBaseType::Named => {
+            if EXTERNAL_FIELD_TYPES.contains(field_typ_name) {
+                match field_typ_name {
+                    "Identity" => quote! { .0 },
+                    "Tai64Timestamp" => quote! { .0.to_le_bytes() },
+                    _ => panic!("From<{field_typ_name}> not implemented for AsRef<u8>."),
+                }
+            } else if !ASREF_BYTE_TYPES.contains(field_typ_name) {
+                quote! { .to_le_bytes() }
+            } else {
+                quote! {}
+            }
+        }
+        FieldBaseType::List => {
+            // TODO: Do we want to allow list types to be hashed into an ID? Would make for
+            // better/greater entropy but would add complexity.
```

Comment:
re: this suggestion -- could we call `to_bytes_tokens on the inner type of the list?